File: src/frontend/lib/dto/payment/payment_request_dto.dart
class PaymentRequestDto {
  final String name;
  final int amount;
  final String currency;

  PaymentRequestDto({
    required this.name,
    required this.amount,
    required this.currency,
  });

  Map<String, dynamic> toJson() => {
    "name": name,
    "amount": amount,
    "currency": currency,
  };
}

File: src/backend/src/main/java/com/bouh/backend/model/Dto/payment/PaymentRequestDto.java
package com.bouh.backend.model.Dto.payment;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor

public class PaymentRequestDto {
    @NotBlank
    private String name;

    @Min(1)
    private Long amount;

    @NotBlank
    private String currency;

}
